### PR TITLE
Added support for selection of branch from GIT

### PR DESCRIPTION
**Pull request by arvindgh**

### CODE REVIEW

**Review by MeniKadosh1 — COMMENTED** [review]
- Jinendra reviewed this code?
- notice that some of the Unit Tests failed on Linux- need to check why
- i didn't noticed in changes that you added the Branch info on code which we creating the Auto Run JSON for the user from the Run set Auto Run Wizard

File: Ginger/Ginger/SourceControl/SourceControlConnDetailsPage.xaml [hunk 1]
```diff
@@ -25,11 +25,15 @@
                 <TextBox x:Name="SourceControlUserAuthorNameTextBox" Style="{StaticResource @TextBoxStyle}" TextWrapping="Wrap" Text="" HorizontalAlignment="Stretch" TextChanged="SourceControlUserDetails_TextChanged"/>
                 <Label Content="Author Email:" Style="{StaticResource @InputFieldLabelStyle}" HorizontalAlignment="Stretch"/>
                 <TextBox x:Name="SourceControlAuthorEmailTextBox" Style="{StaticResource @TextBoxStyle}" TextWrapping="Wrap" Text="" HorizontalAlignment="Stretch" TextChanged="SourceControlUserDetails_TextChanged"/>
+                <StackPanel x:Name="xBranchPanel" Visibility="Visible">
+                    <Label x:Name="xLabelBranch" Content="Source Control Branch:" Style="{StaticResource @InputFieldLabelStyle}" HorizontalAlignment="Stretch"/>
```
Comment (gingergen) [inline Ginger/Ginger/SourceControl/SourceControlConnDetailsPage.xaml:29]: Need to show the branch below the URL field

File: Ginger/Ginger/SourceControl/SourceControlConnDetailsPage.xaml [hunk 2]
```diff
@@ -25,11 +25,15 @@
...
+                    <TextBox x:Name="SourceControlBranchTextBox" Style="{StaticResource @TextBoxStyle}" TextWrapping="Wrap" Text="" HorizontalAlignment="Stretch" IsReadOnly="True" IsEnabled="False"></TextBox>
```
Comment (gingergen) [inline Ginger/Ginger/SourceControl/SourceControlConnDetailsPage.xaml:30]: need to add "x" prefix to all x:Name=

File: Ginger/Ginger/SourceControl/SourceControlProjectsPage.xaml.cs [hunk 3]
```diff
@@ -109,7 +124,7 @@ private void Bind()
             GingerCore.GeneralLib.BindingHandler.ObjFieldBinding(SourceControlUserTextBox, TextBox.TextProperty, mSourceControl, nameof(SourceControlBase.SourceControlUser));
             GingerCore.GeneralLib.BindingHandler.ObjFieldBinding(SourceControlPassTextBox, TextBox.TextProperty, mSourceControl, nameof(SourceControlBase.SourceControlPass));
             GingerCore.GeneralLib.BindingHandler.ObjFieldBinding(txtConnectionTimeout, TextBox.TextProperty, mSourceControl, nameof(SourceControlBase.SourceControlTimeout));
-            UpdateTimeoutVisibility();
+            UpdateTimeoutAndBranchPanelVisibility();
```
Comment (gingergen) [inline Ginger/Ginger/SourceControl/SourceControlProjectsPage.xaml.cs:127]: call the function something like "SetConfigurationsVisibility()" to be more generic

**Review by gingergen — COMMENTED** [review]
see comments inline

**Comment by arvindgh** [comment]
> see comments inline

Done, Handled Review comments.

**Review by MeniKadosh1 — APPROVED** [review]

**Pull request merged**